auacm/angular/src/app/pipes/limit-words.pipe.ts

```diff
@@ -7,18 +8,107 @@ import { Pipe, PipeTransform } from '@angular/core';
   name: 'limitWords'
 })
 export class LimitWordsPipe implements PipeTransform {
+  private converter: Converter = new Converter({'tables': true});
+
+  /**
+   * This transforms a Markdown string into HTML and shortens it based on the amount.
+   *
+   * @param words - the markdown text that needs to be shortened
+   * @param amount - the maximum word count of the returned string
+   */
   transform(words: string, amount: number): string {
+    const htmlWords = this.converter.makeHtml(words);
+    const parsedHtml = document.createElement( 'html' );
+    parsedHtml.innerHTML = htmlWords;
+
+    const tempChildren = parsedHtml.children[1].childNodes;
+    const children = [];
+    for (let i = 0; i < tempChildren.length; i++) {
+      children.push(tempChildren[i]);
+    }
+
+    // let finalHtml = document.createElement( 'html' );
     let finalString = '';
-    const allWords = words.split(' ');
-    let count = 0;
-    for (let i = 0; i < amount && i < allWords.length; i++) {
-      finalString += allWords[i] + ' ';
-      count++;
-    };
-    if (count === amount) {
-      return finalString.trim() + ' ...';
+    let size = 0;
+    for (let i = 0; i < children.length; i++) {
+      const element = this.getStringForElement(children[i], '', size, amount);
+      finalString += element.value;
+      size = element.size;
+    }
+    return finalString;
+  }
+
+  /**
+   * Recursively searches through an HTML element and tries to shorten it to the maxSize.
+   *
+   * @param htmlElement - the element to shorten
+   * @param currentString - the current string that has been shortened
+   * @param currentSize - the size of the currentString
+   * @param maxSize - the maximum amount of words that can be within the currentString
+   * @returns a CustomString containing the shortened element as a string and the size of the element
+   */
+  getStringForElement(htmlElement: any, currentString: string, currentSize: number, maxSize: number): CustomString {
```

Comment:
This method is pretty long. It might be a bit more readable if you broke this into a few helper methods.